docs/proposals/device-management-enhance.md

```diff
@@ -0,0 +1,1069 @@
+	// Customized values for provided protocol
+	// +optional
+	CustomizedValues *CustomizedValue `json:"customizedValues,omitempty"`
+}
+
+type ProtocolConfigTCP struct {
+	// Required.
+	IP string `json:"ip,omitempty"`
+	// Required.
+	Port int64 `json:"port,omitempty"`
+}
+
+type ProtocolConfigCOM struct {
+	...
+	// Required.
+	SerialPort string `json:"serialPort,omitempty"`
+	// Required. BaudRate 115200|57600|38400|19200|9600|4800|2400|1800|1200|600|300|200|150|134|110|75|50
+	Parity string `json:"parity,omitempty"`
+	// Required. Bit that stops 1|2
+	StopBits int64 `json:"stopBits,omitempty"`
+}
+
+type ProtocolConfigCustomized struct {
+	// Unique protocol name
+	// Required.
+	ProtocolName string `json:"protocolName,omitempty"`
+	// Any config data
+	// +optional
+	ConfigData *CustomizedValue `json:"configData,omitempty"`
+}
+
+// DeviceData reports the device's time-series data to edge MQTT broker.
+// These data should not be processed by edgecore. Instead, they can be process by
+// third-party data-processing apps like EMQX kuiper.
+type DeviceData struct {
+	// Required: A list of data properties, which are not required to be processed by edgecore
+	DataProperties []DataProperty `json:"dataProperties,omitempty"`
+	// Topic used by mapper, all data collected from dataProperties
+	// should be published to this topic,
+	// the default value is $ke/events/device/+/data/update
+	// +optional
+	DataTopic string `json:"dataTopic,omitempty"`
+}
+
+// DataProperty represents the device property for external use.
+type DataProperty struct {
+	// Required: The property name for which should be processed by external apps.
+	// This property should be present in the device model.
+	PropertyName string `json:"propertyName,omitempty"`
+	// Additional metadata like timestamp when the value was reported etc.
+	// +optional
+	Metadata map[string]string `json:"metadata,omitempty"`
+}
+
+// DevicePropertyVisitor describes the specifics of accessing a particular device
+// property. Visitors are intended to be consumed by device mappers which connect to devices
+// and collect data / perform actions on the device.
+type DevicePropertyVisitor struct {
+	// Required: The device property name to be accessed. This should refer to one of the
+	// device properties defined in the device model.
+	PropertyName string `json:"propertyName,omitempty"`
+	// Define how frequent mapper will report the value.
+	// +optional
+	ReportCycle int64 `json:"reportCycle,omitempty"`
+	// Define how frequent mapper will collect from device.
+	// +optional
+	CollectCycle int64 `json:"collectCycle,omitempty"`
+	// Customized values for visitor of provided protocols
+	// +optional
+	CustomizedValues *CustomizedValue `json:"customizedValues,omitempty"`
+	// Required: Protocol relevant config details about the how to access the device property.
+	VisitorConfig `json:",inline"`
+}
+```
+
+### modifications on device model types
+```golang
+type VisitorConfig struct {
```

Comment:
Does the VisitorConfig should be move to device instance? Since no one will use it in device model